src/store/channels-list/saga.ts

```diff
@@ -147,7 +148,26 @@ export function* channelsReceived(action) {
   yield put(receive(newChannelList));
 }
 
+function* listenForUserLogin() {
+  const userChannel = yield call(getAuthChannel);
+  while (true) {
+    yield take(userChannel, Events.UserLogin);
+    // Note: not sure why these were tied together. They're separate portions of the application.
+    yield startChannelsAndConversationsRefresh();
+  }
+}
+
+function* listenForUserLogout() {
+  const userChannel = yield call(getAuthChannel);
+  while (true) {
+    yield take(userChannel, Events.UserLogout);
+    yield put({ type: SagaActionTypes.StopChannelsAndConversationsAutoRefresh });
+  }
+}
+
 export function* saga() {
+  yield spawn(listenForUserLogin);
+  yield spawn(listenForUserLogout);
```

Comment:
one doubt here, so this will get "spwaned" & keep listening "forever" in the background? I.e if i login, 169 get's triggered, but if i logout, and then again login, 169 gets fired again, right? (Given that we spawn it a single time from looking at the code)

Reply:
That's correct.